src/PowerShellEditorServices.Hosting/EditorServicesLoader.cs

```diff
@@ -0,0 +1,305 @@
+
+                var asmName = new AssemblyName(args.Name);
+
+                string asmPath = Path.Combine(s_psesDependencyDirPath, $"{asmName.Name}.dll");
+                
+                if (!File.Exists(asmPath))
+                {
+                    return null;
+                }
+
+                logger.Log(PsesLogLevel.Diagnostic, $"Loading {args.Name} from PSES dependency dir into LoadFrom context");
+                return Assembly.LoadFrom(asmPath);
+            };
+#endif
+
+            return new EditorServicesLoader(logger, hostConfig, sessionFileWriter, loggersToUnsubscribe);
+        }
+
+        private readonly EditorServicesConfig _hostConfig;
+
+        private readonly ISessionFileWriter _sessionFileWriter;
+
+        private readonly HostLogger _logger;
+
+        private readonly IReadOnlyCollection<IDisposable> _loggersToUnsubscribe;
+
+        private EditorServicesLoader(
+            HostLogger logger,
+            EditorServicesConfig hostConfig,
+            ISessionFileWriter sessionFileWriter,
+            IReadOnlyCollection<IDisposable> loggersToUnsubscribe)
+        {
+            _logger = logger;
+            _hostConfig = hostConfig;
+            _sessionFileWriter = sessionFileWriter;
+            _loggersToUnsubscribe = loggersToUnsubscribe;
+        }
+
+        /// <summary>
+        /// Load Editor Services and its dependencies in an isolated way and start it.
+        /// This method's returned task will end when Editor Services shuts down.
+        /// </summary>
+        /// <returns></returns>
+        public async Task LoadAndRunEditorServicesAsync()
+        {
+#if !CoreCLR
+            // Make sure the .NET Framework version supports .NET Standard 2.0
+            CheckNetFxVersion();
+#endif
+
+            // Ensure the language mode allows us to run
+            CheckLanguageMode();
+
+            // Add the bundled modules to the PSModulePath
+            UpdatePSModulePath();
+
+            // Log important host information here
+            LogHostInformation();
+
+            // Check to see if the configuration we have is valid
+            ValidateConfiguration();
+
+            // Method with no implementation that forces the PSES assembly to load, triggering an AssemblyResolve event
+            _logger.Log(PsesLogLevel.Verbose, "Loading PSES assemblies");
+            EditorServicesLoading.LoadEditorServicesForHost();
+
+            _logger.Log(PsesLogLevel.Verbose, "Starting EditorServices");
+            using (var editorServicesRunner = EditorServicesRunner.Create(_logger, _hostConfig, _sessionFileWriter, _loggersToUnsubscribe))
+            {
+                // The trigger method for Editor Services
+                // We will wait here until Editor Services shuts down
+                await editorServicesRunner.RunUntilShutdown().ConfigureAwait(false);
+            }
+        }
+
+        public void Dispose()
+        {
+            // TODO: Remove assembly resolve events
```

Comment:
I need to work that particular one out. Currently we resubscribe the host logger to get logging back when we shutdown, so I have to be careful there

Reply:
Oh wait, `_loggersToUnsubscribe` actually covers the loggers that should be unsubscribed *before* the integrated console starts (since at least one of them writes to the host)